numpy/_core/tests/test_multithreading.py

```diff
@@ -30,13 +31,29 @@ def func(seed):
 def test_parallel_ufunc_execution():
     # if the loop data cache or dispatch cache are not thread-safe
     # computing ufuncs simultaneously in multiple threads leads
-    # to a data race
+    # to a data race that causes crashes or spurious exceptions
     def func():
         arr = np.random.random((25,))
         np.isnan(arr)
 
     run_threaded(func, 500)
 
+    # see gh-26690
```

Comment:
More of a testing performance question: this failed only on python 3.13 w/o gil, right? At least, on my linux machine with python 3.11, I don't get errors with the test from the original issue. But I did notice that that test was very slow. If this is similarly slow, it should only be run if relevant, rather than slow every test run down.

Reply:
This test is much faster than the one in the original issue. Is this test still very slow for you? It runs in a fraction of a second on both my test machines.

Reply:
But yes, only free-threaded python 3.13 on linux raises an exception.

Reply:
I didn't test if myself; if it is a fraction of a second, it seems absolutely fine to run it all the time - one never knows the bugs one can catch with it!